### PR TITLE
add update mode for spark writer

### PR DESCRIPTION
**Pull request by Nicole00**
Support two write mode for spark writer： `INSERT` and `UPDATE`.
If the write mode is `INSERT`, then construct the INSERT NGQL for all column of DataFrame except for id(srcId, dstId, rank).
If the write mode is `UPDATE`, then construct the UPDATE NGQL for all column of DataFrame except for id(srcId, dstId, rank). The `UPDATE` mode does not ignore the DataFrame's null value.

### CODE REVIEW

**Comment by Nicole00** [comment]
@cooper-lzy Nebula Spark Connector's document needs to add `UPDATE` write mode.

File: nebula-spark-connector/src/main/scala/com/vesoft/nebula/connector/NebulaConfig.scala [hunk 1]
```diff
@@ -88,12 +88,17 @@ object NebulaConnectionConfig {
 /**
   * Base config needed when write dataframe into nebula graph
   */
-private[connector] class WriteNebulaConfig(space: String, user: String, passwd: String, batch: Int)
+private[connector] class WriteNebulaConfig(space: String,
+                                           user: String,
+                                           passwd: String,
+                                           batch: Int,
+                                           writeMode: String)
```
Comment (HarrisChu) [inline nebula-spark-connector/src/main/scala/com/vesoft/nebula/connector/NebulaConfig.scala:95]: how about using enum for `writeMode`?

File: nebula-spark-connector/src/main/scala/com/vesoft/nebula/connector/Template.scala [hunk 2]
```diff
@@ -15,4 +15,8 @@ object NebulaTemplate {
   private[connector] val EDGE_VALUE_WITHOUT_RANKING_TEMPLATE = "%s->%s: (%s)"
   private[connector] val EDGE_VALUE_TEMPLATE                 = "%s->%s@%d: (%s)"
   private[connector] val USE_TEMPLATE                        = "USE %s"
+
+  private[connector] val UPDATE_VERTEX_TEMPLATE = "UPDATE %s ON %s %s SET %s;"
+  private[connector] val UPDATE_EDGE_TEMPLATE   = "UPDATE %s ON %s %s->%s@%d SET %s;"
```
Comment (HarrisChu) [inline nebula-spark-connector/src/main/scala/com/vesoft/nebula/connector/Template.scala:20]: escape keywords like 
```
UPDATE VERTEX ON `Tag` "player101" SET `Tag`=`Tag`+2`
```

File: nebula-spark-connector/src/main/scala/com/vesoft/nebula/connector/NebulaConfig.scala (hunk 1 above)
Comment (Nicole00) [reply nebula-spark-connector/src/main/scala/com/vesoft/nebula/connector/NebulaConfig.scala:95]: Thanks, entry parameter is changed to WriteMode enum.

**Review by HarrisChu — APPROVED** [review]

**Pull request merged**